feat(gateway): Split vault providers into their own pages (#6454)
Kong/developer.konghq.com: app/gateway/vault/azure.md

---
title: "Azure Key Vault"
layout: reference
content_type: reference
description: "Use Azure Key Vault to store and reference secrets in {{site.base_gateway}}."
breadcrumbs:
  - /gateway/
  - /gateway/entities/
  - /gateway/entities/vault/

permalink: /gateway/entities/vault/azure/

works_on:
  - on-prem
  - konnect

products:
  - gateway

tools:
  - admin-api
  - konnect-api
  - deck
  - kic
  - terraform

tags:
  - secrets-management
  - azure

min_version:
  gateway: '3.5'

related_resources:
  - text: "{{site.base_gateway}} Vault entity"
    url: /gateway/entities/vault/
  - text: "Supported {{site.base_gateway}} Vault backends"
    url: /gateway/entities/vault/#supported-vault-backends
  - text: Secrets management
    url: /gateway/secrets-management/
  - text: Azure Key Vault documentation
    url: https://learn.microsoft.com/azure/key-vault/
---

You can set up an [Azure Key Vault](https://learn.microsoft.com/azure/key-vault/) vault in one of the following ways:
* Using the [Vault entity](/gateway/entities/vault/)

> * Using [environment variables](/gateway/manage-kong-conf/#environment-variables), set at {{site.base_gateway}} startup

Reviewer: ```suggestion
* Using [environment variables](/gateway/entities/vault/#store-secrets-as-environment-variables), set at {{site.base_gateway}} startup
```

* Using parameters in [`kong.conf`](/gateway/configuration/), set at {{site.base_gateway}} startup

The Vault entity can only be used once the database is initialized.
Secrets for values that are used before the database is initialized can't make use of the Vaults entity.

## Azure Key Vault authentication

{{site.base_gateway}} uses a key to automatically authenticate
with the [Azure Key Vaults API](https://learn.microsoft.com/en-us/rest/api/keyvault/) and grant you access.
If you're using a client secret for authentication, set the following environment variable on your data plane to connect with an Azure Key Vault:

```bash

> export AZURE_CLIENT_SECRET=YOUR_CLIENT_SECRET

Reviewer: ```suggestion
export AZURE_CLIENT_SECRET='YOUR_CLIENT_SECRET'
```

```
If you're using an Instance Managed Identity Token, you don't need to set the client secret env variable.

{% include_cached /gateway/azure-vault-dcgw-unsupported.md %}

At minimum, you'll also need to set the following values on your data plane.

```sh
export KONG_VAULT_AZURE_VAULT_URI=https://your-vault.vault.azure.com
export KONG_VAULT_AZURE_TENANT_ID=YOUR_TENANT_ID

> export KONG_VAULT_AZURE_CLIENT_ID=YOUR_CLIENT_ID

Reviewer: ```suggestion
export KONG_VAULT_AZURE_VAULT_URI='https://your-vault.vault.azure.com'
export KONG_VAULT_AZURE_TENANT_ID='YOUR_TENANT_ID'
export KONG_VAULT_AZURE_CLIENT_ID='YOUR_CLIENT_ID'
```

<!--vale off-->
{% table %}
columns:
  - title: Field name
    key: field
  - title: Parameter format
    key: parameter
  - title: Description
    key: description
rows:
  - field: Vault URI
    parameter: |
      * **Vault entity:** `vaults.config.vault_uri`
      * **kong.conf parameter:** `vault_azure_vault_uri`
      * **Environment variable:** `KONG_VAULT_AZURE_VAULT_URI`
    description: |
      The URI from which the vault is reachable. This value can be found in your Azure Key Vault Dashboard under the Vault URI entry.
  - field: Client ID
    parameter: |
      * **Vault entity:** `vaults.config.client_id`
      * **kong.conf parameter:** `vault_azure_client_id`
      * **Environment variable:** `KONG_VAULT_AZURE_CLIENT_ID`
    description: |
      The client ID for your registered application. You can find this in the Azure Dashboard under App Registrations.
  - field: Tenant ID
    parameter: |
      * **Vault entity:** `vaults.config.tenant_id`
      * **kong.conf parameter:** `vault_azure_tenant_id`
      * **Environment variable:** `KONG_VAULT_AZURE_TENANT_ID`
    description: |

>       The `DirectoryId` and `TenantId` are the same: both refer to the GUID representing your Azure Active Directory tenant. Microsoft documentation and products may use either term depending on context.

Reviewer: ```suggestion
      The `DirectoryId` and `TenantId` are the same: both refer to the GUID representing your Azure Entra tenant. Microsoft documentation and products may use either term depending on context.
```
